### PR TITLE
[DOCS-7193] Add SDS Regex Syntax doc

### PR DESCRIPTION
**Pull request by maycmlee**
### What does this PR do? What is the motivation?

Adds a new doc that lists the supported regex syntax for creating custom Sensitive Data Scanner rules.
Adds a new entry to the nav.

[DOCS-7193](https://datadoghq.atlassian.net/browse/DOCS-7193)

### Merge instructions

- [ ] Please merge after reviewing

### Additional notes

[DOCS-7193]: https://datadoghq.atlassian.net/browse/DOCS-7193?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: content/en/sensitive_data_scanner/regular_expression_syntax.md [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+
+| Unicode Properties| Description           |
+| ------------------| --------------------- |
+| `C`               | Other                 |
+| `Cc`              | Control               |
+| `Cf`              | Format                |
+| `Cn`              | Unassigned            |
+| `Co`              | Private use           |
+| `Cs`              | Surrogate             |
+|                   |                       |
```
Comment (fuchsnj) [inline content/en/sensitive_data_scanner/regular_expression_syntax.md:183]: The empty rows seemed a bit odd to me at first, I thought some values were missing. I'd personally prefer to just not have the extra spacing unless this is common in other places (or maybe there's another way to add a separator?). Just a preference here, feel free to ignore it.

**Review by fuchsnj — APPROVED** [review]
Overall this looks good, thanks! Just some minor comments / nits.

File: content/en/sensitive_data_scanner/regular_expression_syntax.md [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+## Character escapes
+
+| Regex Syntax     | Description                                                         |
+| --------------- | -------------------------------------------------------------------- |
+| `\xhh`          | Escapes characters with hex code `hh` (up to 2 digits are allowed).  |
+| `\x{hhhhhh}`    | Escapes characters with hex code `hhhhhh` (between 1 and 6 digits).  |
+| `\a`            | Escapes a bell `(\x{7})`.                                            |
+| `\b`            | Escapes a backspace `(\x{8})`. This only works in a custom character class (for example, `[\b]`), otherwise it is treated as a word boundary. |
+| `\cx`           | Escapes a control sequence, where `x` is `A-Z` (upper or lowercase). For example: `\cA` = `\x{0}`, `\cB` = `\x{1}`,...`\cZ` = `\x{19}`. |
+| `\e`            | Escapes `(\x{1B})`.                                                  |
```
[the remark this answers is not in the document]
Comment (maycmlee) [reply content/en/sensitive_data_scanner/regular_expression_syntax.md]: Thanks, updated!

File: content/en/sensitive_data_scanner/regular_expression_syntax.md [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+
+## Custom character classes
+
+| Regex Syntax                    | Description                                                                         |
+| ------------------------------- | ----------------------------------------------------------------------------------- |
+|`[...]`                          | Matches any character listed inside the brackets.                                   |
+| `[^...]`                        | Matches anything that is not listed inside the brackets.                            |
+| `[a-zA-Z]`                      | Matches anything in the range `A - Z` (upper or lowercase).                         |
+| `[\s\w\d\S\W\D\v\V\h\H\p{x}...]`| Other classes defined above are allowed (except `.` which is treated as a literal). |
+| `[[:ascii_class:]]`             | Matches special named [ASCII classes](#ascii-classes).                              |
```
[the remark this answers is not in the document]
Comment (maycmlee) [reply content/en/sensitive_data_scanner/regular_expression_syntax.md:112]: Updated

File: content/en/sensitive_data_scanner/regular_expression_syntax.md [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+| ------------------------------------------------------ | ---------------------- |
+| `(...)`                                                | A capture group.       |
+| `(?<name>...)`                                         | A named capture group. |
+| `(?P<name>...)`                                        | A named capture group. |
+| `(?'name'...)`                                         | A named capture group. |
+| `(?:...)`                                              | A non-capturing group. |
+
+## Setting flags
+
+Use flags to modify regex match. There are two ways to specify flags:
```
[the remark this answers is not in the document]
Comment (maycmlee) [reply content/en/sensitive_data_scanner/regular_expression_syntax.md]: Thanks! Updated to "Use flags to modify the regex behavior."

File: content/en/sensitive_data_scanner/regular_expression_syntax.md (hunk 1 above)
Comment (maycmlee) [reply content/en/sensitive_data_scanner/regular_expression_syntax.md]: Ah makes sense, I removed them. We don't do that anywhere else in the docs afaik.

File: content/en/sensitive_data_scanner/regular_expression_syntax.md [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+---
+
+## Overview
+
+In [Sensitive Data Scanner][1], the scanning rule determines what sensitive information to match within the data. You can use rules from the [Scanning Rule Library][2] or you can create custom scanning rules using regular expression (regex) patterns to scan for sensitive information. The Sensitive Data Scanner regex syntax is a subset of [PCRE2][3].
+
+{{< whatsnext desc="The available regex syntax for custom rules are broken down into the following categories:" >}}
+    {{< nextlink href="sensitive_data_scanner/regular_expression_syntax#alternation" >}}Alternation{{< /nextlink >}}
+    {{< nextlink href="sensitive_data_scanner/regular_expression_syntax#assertions" >}}Assertions{{< /nextlink >}}
+    {{< nextlink href="sensitive_data_scanner/regular_expression_syntax#ascii_classes" >}}ASCII classes{{< /nextlink >}}
```
Comment (estherk15) [inline content/en/sensitive_data_scanner/regular_expression_syntax.md:18]:
```suggestion
    {{< nextlink href="sensitive_data_scanner/regular_expression_syntax#ascii-classes" >}}ASCII classes{{< /nextlink >}}
```

**Review by estherk15 — APPROVED** [review]
Looks good! Do you think this page will grow? Would it be worth breaking each out to its own page-something to think about for future

File: content/en/sensitive_data_scanner/regular_expression_syntax.md (hunk 5 above)
Comment (maycmlee) [reply content/en/sensitive_data_scanner/regular_expression_syntax.md:18]: Thanks for the catch! I don't think they plan on making more regex syntax available.

**Pull request merged**